debugger/src/debug_adapter/protocol.rs

```diff
@@ -63,11 +63,21 @@ impl<R: Read, W: Write> DapAdapter<R, W> {
         let response_header = format!("Content-Length: {}\r\n\r\n", response_body.len());
 
         self.output.write_all(response_header.as_bytes())?;
+        self.output.flush()?;
```

Comment:
What's the reason for the flush here?

A single call to flush should be enought at the end.

Reply:
With a large payload in the body, I consistently errorred with "WouldBlock". Once I added it, the error went away.

Reply:
I think I figured out what's going on, we set the socket to non-blocking in the startup:https://github.com/probe-rs/probe-rs/blob/ff7c570cae29c32adbf3026b19990f6ebd941bb7/debugger/src/server/startup.rs#L88

And that leads to the issue here, were the `flush` is needed to wait until all data is written. If the data is not flushed, the data will just be in the buffer, and then the next call to `write_all` will return `WouldBlock`.

I don't mind using `flush` for making the socket effectively blocking again, but I'm a bit worried about the arbitrary value of 1024 for the chunk size. That looks like something which could be influenced by the OS, so it might behave differently on different OSes. 

I think it would be better to just set the socket to blocking, if that is the behaviour we want. That probably requires some changes for the receiving code, but should be possible.

Reply:
@Tiwalun The 'non-blocking' was implemented to allow us to do retries on reading data from the client : https://github.com/probe-rs/probe-rs/blob/ff7c570cae29c32adbf3026b19990f6ebd941bb7/debugger/src/debug_adapter/protocol.rs#L193-L207

It allows us to detect the client has no data ready, and then go back to polling probe status, etc. until the next loop where we check if the client has data for us. 

Can you align with leaving this code as-is, since it is working, and if we ever run into specific errors relating to the 1024 chunk size, then we can deal with it then?

Reply:
I'd prefer an implementation with just `write`, but if that's too much effort the current workaround is fine as well, I guess...

`write_all` is just a call to `write` in a loop, so using `write` instead will work as well, and would avoid the hardcoded chunk size.

Reply:
@Tiwalun, I've just checked in the change you requested. I hope this is satisfactory.

Reply:
Looks good, thanks a lot for changing it. I'm more comfortable with this solution.